### DISCUSSION TITLE
Why os.date() and table.concat() is not safe function

### DISCUSSION THREAD

**Discussion by GYWang1983** [Kong/kong#8071] [Help] [answered]
```lua
-- Safe packages/functions below
```
https://github.com/Kong/kong/blob/4f5b85dcdba5ff660a758bc001073992fb265ebc/kong/tools/kong-lua-sandbox.lua#L73
`os.date` is useful for formatting timestamp to date string in http headers

https://github.com/Kong/kong/blob/4f5b85dcdba5ff660a758bc001073992fb265ebc/kong/tools/kong-lua-sandbox.lua#L79

Use `table.concat` instead of `..` is best practice in openresty, and there have been too many codes use the function to concat string in pre-function or post-function

**Comment by kikito** [comment, marked as answer]
Thanks for this report. I have studied both functions and I see no reason why they should be blocked by the sandbox. I have created a PR for including those https://github.com/Kong/kong/pull/8081
